### PR TITLE
Fixed multiple bugs in `yii\db\Query::getTablesUsedInFrom()`

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #15356. Replaces #15395

### CODE REVIEW

File: framework/db/Query.php [hunk 1]
```diff
@@ -466,18 +466,29 @@ public function getTablesUsedInFrom()
...
         if (empty($this->from)) {
             return [];
-        } elseif (is_array($this->from)) {
+        }
+
+        if (is_array($this->from)) {
             $tableNames = $this->from;
         } elseif (is_string($this->from)) {
             $tableNames = preg_split('/\s*,\s*/', trim($this->from), -1, PREG_SPLIT_NO_EMPTY);
+        } elseif ($this->from instanceof Expression) {
```
Comment (SilverFire) [inline framework/db/Query.php]: Should it be replaced with `ExpressionInterface` after merging that PR?

File: framework/db/Query.php [hunk 2]
```diff
@@ -466,18 +466,29 @@ public function getTablesUsedInFrom()
...
 
         // Clean up table names and aliases
         $cleanedUpTableNames = [];
         foreach ($tableNames as $alias => $tableName) {
-            if (!is_string($alias)) {
+            $tableNameString = null;
+            if (is_string($tableName)) {
+                $tableNameString = $tableName;
+            } elseif ($tableName instanceof Expression) {
+                $tableNameString = $tableName->expression;
```
Comment (SilverFire) [inline framework/db/Query.php]: Again, in terms of that PR, this code support only `Expression` class, or any `ExpressionInterface`?

File: framework/db/Query.php [hunk 3]
```diff
@@ -507,32 +520,42 @@ public function getTablesUsedInFrom()
...
+                        $tableNameString = $alias = $matches[1];
                     }
                 }
             }
 
-            $tableName = str_replace(["'", '"', '`', '[', ']'], '', $tableName);
-            $alias = str_replace(["'", '"', '`', '[', ']'], '', $alias);
-
-            $cleanedUpTableNames[$alias] = $tableName;
+            if ($tableName instanceof Expression) {
```
Comment (SilverFire) [inline framework/db/Query.php]: Same here

**Review by SilverFire — APPROVED** [review]
Good otherwise

File: framework/db/Query.php (hunk 1 above)
Comment (samdark) [reply framework/db/Query.php]: Yes.

File: framework/db/Query.php (hunk 2 above)
Comment (samdark) [reply framework/db/Query.php]: `ExpressionInterface` isn't in master yet.

Comment (samdark) [reply framework/db/Query.php]: I think in this case it's about class.

**Comment by samdark** [comment]
Thank you @vladis84 for starting it.

**Pull request merged**